src/runtime/test_case_runner.js

```diff
@@ -199,7 +212,13 @@ export default class TestCaseRunner {
 
   async runSteps() {
     await Promise.each(this.testCase.pickle.steps, async step => {
+      // TODO handle failure
+      await this.runStepHooks(this.beforeStepHookDefinitions, {
+        sourceLocation: this.testCaseSourceLocation,
+        pickle: this.testCase.pickle,
```

Comment:
Hmm. I don't expect the beforeStep would need these parameters. I think we could start with passing in only the step. The after step will end up getting the step and the stepResult.